dbt/adapters/base/impl.py

```diff
@@ -300,12 +300,13 @@ def behavior(self) -> Behavior:
     @behavior.setter  # type: ignore
     def behavior(self, flags: List[BehaviorFlag]) -> None:
         flags.extend(self._behavior_flags)
-        try:
-            # we don't always get project flags, for example during `dbt debug`
-            self._behavior = Behavior(flags, self.config.flags)
-        except AttributeError:
-            # in that case, don't load any behavior to avoid unexpected defaults
-            self._behavior = Behavior([], {})
+
+        # we don't always get project flags, for example, the project file is not loaded during `dbt debug`
+        # in that case, load the default values for behavior flags to avoid compilation errors
+        # this mimics not loading a project file, or not specifying flags in a project file
+        user_overrides = getattr(self.config, "flags", {})
```

Comment:
This is effectively the same as swapping `[]` for `flags` on line `308`, but doing this reads much more cleanly.